compiler/src/iree/compiler/Bindings/Native/Transforms/ConvertStreamableOps.cpp

```diff
@@ -0,0 +1,385 @@
+  SmallVector<Value> resultDims;
+  if (streamableFunc.resultDimsFunc) {
+    // Call the custom shape calculation function, if specified.
+    // It should return the required number of dynamic dimensions.
+    SmallVector<Type> resultDimTypes(streamableFunc.requiredResultDims,
+                                     builder.getIndexType());
+    auto calculateCallOp = builder.create<func::CallOp>(
+        callOp.getLoc(), streamableFunc.resultDimsFunc, resultDimTypes,
+        callOp.getOperands());
+    llvm::append_range(resultDims, calculateCallOp.getResults());
+  } else {
+    // Get the shape dimensions from existing call arguments or tied operands.
+    for (auto [i, resultType] : llvm::enumerate(callOp.getResultTypes())) {
+      if (auto shapedType = resultType.dyn_cast<ShapedType>()) {
+        const auto &resultDimArgs = streamableFunc.resultDimArgs[i];
+        if (resultDimArgs.empty()) continue;
+        if (resultDimArgs.front() == kTiedDim) {
+          // Source from a tied operand. Types must match exactly.
+          assert(streamableFunc.tiedOperands[i] !=
+                     IREE::Util::TiedOpInterface::kUntiedIndex &&
+                 "tied dims must be from tied operands");
+          auto sourceArg = callOp.getOperand(streamableFunc.tiedOperands[i]);
+          assert(sourceArg.getType() == resultType &&
+                 "only valid to infer result shapes from identically typed "
+                 "tied operands");
+          llvm::append_range(resultDims,
+                             IREE::Util::buildDynamicDimsForValue(
+                                 callOp.getLoc(), sourceArg, builder));
+        } else {
+          // Source from call arguments.
+          for (int64_t j : resultDimArgs) {
+            resultDims.push_back(callOp.getOperand(j));
+          }
+        }
+      }
+    }
+  }
+
+  // Replace the original func.call with the new flow.call.
+  auto streamableCallOp = builder.create<IREE::Flow::CallOp>(
+      callOp.getLoc(), callOp.getCalleeAttr(), callOp.getResultTypes(),
+      resultDims, callOp.getOperands(), argDims,
+      streamableFunc.funcOp.getTiedOperandsAttr());
+  streamableCallOp->setDialectAttrs(callOp->getDialectAttrs());
+  callOp.replaceAllUsesWith(streamableCallOp.getResults());
+  callOp.erase();
+
+  return success();
+}
+
+static LogicalResult convertStreamableCalls(
+    mlir::ModuleOp moduleOp,
+    DenseMap<StringRef, StreamableFunc> &streamableFuncs) {
+  return moduleOp.walk([&](func::CallOp callOp) {
+                   auto it = streamableFuncs.find(callOp.getCallee());
+                   if (it != streamableFuncs.end()) {
+                     if (failed(convertStreamableCall(it->second, callOp))) {
+                       return WalkResult::interrupt();
+                     }
+                   }
+                   return WalkResult::advance();
+                 })
+                 .wasInterrupted()
+             ? failure()
+             : success();
```

Comment:
Style nit: break into a variable to save a bit on indentation and line count?
```suggestion
  auto walkResult = moduleOp.walk([&](func::CallOp callOp) {
    auto it = streamableFuncs.find(callOp.getCallee());
    if (it != streamableFuncs.end()) {
      if (failed(convertStreamableCall(it->second, callOp))) {
        return WalkResult::interrupt();
      }
    }
    return WalkResult::advance();
  });
  return walkResult.wasInterrupted() ? failure() : success();
```